### ISSUE TITLE
BHoM_Adapter: Type requests do not work without FilterRequest wrapper when wrapped into ILogicalRequest

### ISSUE THREAD

**Issue by pawelbaran** [BHoM/BHoM_Adapter#241]
#### Description:

That works:

![image](https://user-images.githubusercontent.com/26874773/87783909-aa0caa00-c835-11ea-8903-e93abf5ed5b7.png)

That does not:

![image](https://user-images.githubusercontent.com/26874773/87784021-d9bbb200-c835-11ea-8bc4-51b050a46bda.png)

It could be simply resolved with a `Create.LogicalNotRequest` method that would take `Type` and wrap it into a `FilterRequest`, however I would like to first check with if that is not too much of a crowbar solution, therefore raised in the BHoM_Adapter. What do you think @alelom?

**Comment by alelom** [comment]
I would be happy to add its support if we make the case that the LogicalNotRequest is in the base oM.
So far we have hardcoded support only for FilterRequest because it was deemed "universal". If we make an overhaul of the requests - where the logicalNot becomes one of the basic ones, together with then new FilterRequest (--> to become TypeRequest), then sure, we can add support.
But I'd classify this as feature, it's not a bug - BHoM adapter could not know of LogicalNot.

**Comment by pawelbaran** [comment]
Right, good point about the Feature. When should we expect the overhaul of requests? Would be good to map out handling of `ILogicalRequests` as a part of it.

**Comment by alelom** [comment]
Not sure - wasn't you that created the concept of `ILogicalRequest` (which by the way I like a lot) in revit?
You could simply make the case for moving the base interface in the oM, and defining the specific guidelines for that. 
Then it is a matter of adding the support in the adapter.

**Comment by pawelbaran** [comment]
It already is in Data_oM together with `Create` methods in Data_Engine.
